### PR TITLE
autocomplete: fix suggestions list

### PR DESCRIPTION
**Pull request by jma**
Old version of the third party library `ngx-bootstrap` had sometime wired
behaviors. A hack to avoid a form submission when a suggestion
selected with a return key has been removed. Moreover, the autocomplete
can use angular navigation to go to a record detailed view, the
corresponding detailed view component should observe the route to change
his content when the url change.

* Use `ngx-bootstrap` version 3.6.2.
* Remove an useless hack due to an old version of `ngx-bootstrap`.
* Enables route change detection for the record detailed view.
* Fixes `ngx-bootstrap` imports to be angular 9 and `ngx-bootstrap>6`
compliant.

Note: several versions of `ngx-bootstrap` has been tried, but some of
them failed:
- 3.3.x: does not display a `typeahead` suggestion list for slow query.
- 3.6.1: does not compile our project in production.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Required by
- rero-ils-ui: https://github.com/rero/rero-ils-ui/pull/260

## Why are you opening this PR?

- To fix the autocomplete component.

## How to test?

- Basically needs to test all the widget used by the library

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/autocomplete/autocomplete.component.ts [hunk 1]
```diff
@@ -187,6 +187,7 @@ export class AutocompleteComponent implements OnInit {
             )
           );
         });
+        console.log(values);
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/autocomplete/autocomplete.component.ts:190]: Remove console.log
